src/components/PurchaseOrder/util.js

```diff
@@ -43,6 +43,8 @@ export const isReceiveAvailableForOrder = (order = {}) => {
   return hasLineItemsToReceive && isWorkflowStatusNotPending(order);
 };
 
+export const isRequiredWithFieldValue = (form, field, value) => get(form, field, '') === value;
```

Comment:
What do you think of creating simple function `isMaterialTypeRequired()` ? Basically, business logic is already implemented inside `requiredMaterialType`, so it would be great to have a separate function and call it in both places